README.md

```diff
@@ -135,6 +137,20 @@ fn bind_and_accept(addr: SocketAddr)
 }
 ```
 
+## Pinning
+
+If you wrap the stream in `Box::pin`, the callers won't need to pin it before iterating it. This is more convenient for the callers, but adds a cost of using dynamic dispatch.
```

Comment:
Would be good to explain that `Unpin` gets auto added in `impl Trait` just like how `Send` gets leaked since that could be confusing.